src/lib/device-types/sync.ts

```diff
@@ -0,0 +1,95 @@
+import * as _ from 'lodash';
+import Bluebird = require('bluebird');
+import { captureException } from '../../platform/errors';
+import { sbvrUtils, permissions } from '@balena/pinejs';
+import { getAllDeviceTypes } from '.';
+import { DeviceType } from './build-info-facade';
+
+const DEVICE_TYPES_CACHE_EXPIRATION = 5 * 1000; // 5 mins
+const RETRY_WAIT_TIME = 10 * 1000;
+
+type DeviceTypeResourceKeys = 'slug' | 'name' | 'is_private';
+
+// This map will hold information on which device type fields imported from the device type registry will be synced to which db fields.
+export const syncSettings: {
+	map: {
+		[k in DeviceTypeResourceKeys]?: {
+			name: keyof DeviceType;
+			default?: any;
+		};
+	};
+} = {
+	map: { slug: { name: 'slug' } },
+};
+
+export function setSyncMap(map: typeof syncSettings['map']) {
+	syncSettings.map = _.merge(map, syncSettings.map);
+}
+
+const mapModel = (deviceType: DeviceType, map: typeof syncSettings['map']) => {
+	return Object.keys(map).reduce(
+		(
+			mappedModel: { [k in DeviceTypeResourceKeys]?: any },
+			key: Partial<DeviceTypeResourceKeys>,
+		) => {
+			mappedModel[key] = deviceType[map[key]!.name] ?? map[key]!.default;
+			return mappedModel;
+		},
+		{},
+	);
```

Comment:
I think it's clearer and more concise for a straightforward for loop:
```suggestion
	const mappedModel: { [k in DeviceTypeResourceKeys]?: any } = {};
	for (const key of Object.keys(map)) {
		mappedModel[key] = deviceType[map[key]!.name] ?? map[key]!.default;
	}
	return mappedModel;
```